cmd/chartsvc/handler.go

```diff
@@ -444,3 +445,75 @@ func newChartVersionListResponse(c *models.Chart) apiListResponse {
 
 	return cvl
 }
+
+func handleRedirect(w http.ResponseWriter, req *http.Request) {
+
+	// The URL can be in one of two forms:
+	// - /v1/redirect/charts/stable/aerospike
+	// - /v1/redirect/charts/stable/aerospike/v1.2.3
```

Comment:
I would split the handlers up and extract the shared logic to other functions. We should try to use the router as much as possible to handle path parameters.

Reply:
So... here is my problem and maybe you can help with it. Let's say I have the path `my-repo/my-chart/0.1.0/`. Now we also need to get the provenance file for that. This would be `my-repo/my-chart/0.1.0/.prov`. Notice the placements of the path separators. Or, what if we have the path `my-repo/my-chart/0.1.0` and need to get `my-repo/my-chart/0.1.0.prov`. How do we separate the chart tarball from the provenance one with mux?

There is some variation to the possible inputs which makes handling some of this in the router difficult. Any pointers on that?